### PR TITLE
[npu] fix arange dtype for npu

### PR DESCRIPTION
**Pull request by YanhuiDua**
### PR types
Others

### PR changes
APIs

### Description
npu arange kernel don't support fp16, and it can't be fallbacked to cpu

### CODE REVIEW

File: paddlenlp/transformers/chatglm/modeling.py [hunk 1]
```diff
@@ -117,7 +117,13 @@ def forward(self, position_ids):
...
 
             # x.shape = [b, s, n, h/n/2]
-            t = paddle.arange(seq_len, dtype=self.inv_freq.dtype)
+            # TODO(duanyanhui): npu arange kernel don't support fp16, and
+            # it can't be fallbacked to cpu. It will be fixed in future.
+            if paddle.get_device().split(":")[0] == "npu":
+                t = paddle.arange(seq_len, "float32")
+                t = t.cast(self.inv_freq.dtype)
+            else:
+                t = paddle.arange(seq_len, self.inv_freq.dtype)
```
Comment (sijunhe) [inline paddlenlp/transformers/chatglm/modeling.py]:
```suggestion
               t = paddle.arange(start=0, end=seq_len, dtype=self.inv_freq.dtype)
```

File: paddlenlp/transformers/chatglm/modeling.py [hunk 2]
```diff
@@ -117,7 +117,13 @@ def forward(self, position_ids):
             self.max_seq_len_cached = seq_len
 
             # x.shape = [b, s, n, h/n/2]
-            t = paddle.arange(seq_len, dtype=self.inv_freq.dtype)
+            # TODO(duanyanhui): npu arange kernel don't support fp16, and
+            # it can't be fallbacked to cpu. It will be fixed in future.
+            if paddle.get_device().split(":")[0] == "npu":
+                t = paddle.arange(seq_len, "float32")
```
Comment (sijunhe) [inline paddlenlp/transformers/chatglm/modeling.py]:
```suggestion
                t = paddle.arange(start=0, end=seq_len, dtype="float32")
```

**Review by sijunhe — COMMENTED** [review]
单测有点问题，使用kv args应该可以fix

File: paddlenlp/transformers/chatglm/modeling.py (hunk 2 above)
Comment (YanhuiDua) [reply paddlenlp/transformers/chatglm/modeling.py]: done, thx

**Review by sijunhe — APPROVED** [review]
lgtm

**Pull request merged**